zerver/lib/actions.py

```diff
@@ -1364,8 +1364,9 @@ def do_send_messages(messages_maybe_none: Sequence[Optional[MutableMapping[str,
 
         # Add members of the mentioned user groups into `mentions_user_ids`.
         for group_id in message['message'].mentions_user_group_ids:
-            members = message['mention_data'].get_group_members(group_id)
-            message['message'].mentions_user_ids.update(members)
+            ids = message['mention_data'].get_group_members(group_id)
+            group_mentions = {id: group_id for id in ids if id not in message['message'].mentions_user_ids}
```

Comment:
Stylistically, we don't generally write this sort of list comprehension one-liner; it's less readable.